### DISCUSSION TITLE
📚 Documentation: size limits

### DISCUSSION THREAD

**Discussion by DankMemeGuy** [appwrite/appwrite#4838] [Q&A] [answered]
### 💭 Description

I am reading the documentation and there doesn't seem to be limits listed, or best practices. What are the limits of the various features?

1. How large can a database get? I have a list of 50-100 million rows, can a database hold that?
2. How many files can be uploaded? is there a physical size limit? GB TB? Is there a quantity limit? 10,000 files, 50,000 files etc?
3. How many users can it handle?
4. etc

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
At the moment, Appwrite uses MariaDB as the underlying database. Each project's users list, collection, and bucket is a separate table in the database. I believe we're using the [default 16KB page size](https://mariadb.com/kb/en/innodb-system-variables/#innodb_page_size), which means the maximum table size is [64TB](https://mariadb.com/kb/en/innodb-limitations/#limitations-on-size). Realistically, you'll probably run into other problems if you're using 1 mariadb server before you hit these limits.
